### ISSUE TITLE
Server crash because of Invalid line in auth.txt" from "/home/quentinbd/mff-hg/bin/../builtin/game/auth.lua"

### ISSUE THREAD

**Issue by BetterToAutomateTheWorld** [MinetestForFun/server-minetestforfun-hungry_games#88]
#### Server crash because of "Invalid line in auth.txt" from "/home/quentinbd/mff-hg/bin/../builtin/game/auth.lua"

```
----------------------
Server restarted at samedi 26 mars 2016, 15:59:35 (UTC+0100)
----------------------
Using game specified by --gameid on the command line
2016-03-26 15:59:35: ERROR[main]: ========== ERROR FROM LUA ===========
2016-03-26 15:59:35: ERROR[main]: Failed to load and run script from 
2016-03-26 15:59:35: ERROR[main]: /home/quentinbd/mff-hg/bin/../builtin/init.lua:
2016-03-26 15:59:35: ERROR[main]: /home/quentinbd/mff-hg/bin/../builtin/game/auth.lua:48: Invalid line in auth.txt: "leonidos:8JKPyY0sgCI+wCkoH9x"
2016-03-26 15:59:35: ERROR[main]: stack traceback:
2016-03-26 15:59:35: ERROR[main]:   [C]: in function 'error'
2016-03-26 15:59:35: ERROR[main]:   /home/quentinbd/mff-hg/bin/../builtin/game/auth.lua:48: in function 'read_auth_file'
2016-03-26 15:59:35: ERROR[main]:   /home/quentinbd/mff-hg/bin/../builtin/game/auth.lua:82: in main chunk
2016-03-26 15:59:35: ERROR[main]:   [C]: in function 'dofile'
2016-03-26 15:59:35: ERROR[main]:   /home/quentinbd/mff-hg/bin/../builtin/game/init.lua:19: in main chunk
2016-03-26 15:59:35: ERROR[main]:   [C]: in function 'dofile'
2016-03-26 15:59:35: ERROR[main]:   /home/quentinbd/mff-hg/bin/../builtin/init.lua:25: in main chunk
2016-03-26 15:59:35: ERROR[main]: ======= END OF ERROR FROM LUA ========
2016-03-26 15:59:35: ERROR[main]: ERROR: An unhandled exception occurred: ModError: Failed to load and run /home/quentinbd/mff-hg/bin/../builtin/init.lua
2016-03-26 15:59:35: ERROR[main]: Error from Lua:
2016-03-26 15:59:35: ERROR[main]: /home/quentinbd/mff-hg/bin/../builtin/game/auth.lua:48: Invalid line in auth.txt: "leonidos:8JKPyY0sgCI+wCkoH9x"
2016-03-26 15:59:35: ERROR[main]: stack traceback:
2016-03-26 15:59:35: ERROR[main]:   [C]: in function 'error'
2016-03-26 15:59:35: ERROR[main]:   /home/quentinbd/mff-hg/bin/../builtin/game/auth.lua:48: in function 'read_auth_file'
2016-03-26 15:59:35: ERROR[main]:   /home/quentinbd/mff-hg/bin/../builtin/game/auth.lua:82: in main chunk
2016-03-26 15:59:35: ERROR[main]:   [C]: in function 'dofile'
2016-03-26 15:59:35: ERROR[main]:   /home/quentinbd/mff-hg/bin/../builtin/game/init.lua:19: in main chunk
2016-03-26 15:59:35: ERROR[main]:   [C]: in function 'dofile'
2016-03-26 15:59:35: ERROR[main]:   /home/quentinbd/mff-hg/bin/../builtin/init.lua:25: in main chunk
----------------------
Server restarted at samedi 26 mars 2016, 16:00:00 (UTC+0100)
----------------------
Using game specified by --gameid on the command line
2016-03-26 16:00:00: ERROR[main]: ========== ERROR FROM LUA ===========
2016-03-26 16:00:00: ERROR[main]: Failed to load and run script from 
2016-03-26 16:00:00: ERROR[main]: /home/quentinbd/mff-hg/bin/../builtin/init.lua:
2016-03-26 16:00:00: ERROR[main]: /home/quentinbd/mff-hg/bin/../builtin/game/auth.lua:48: Invalid line in auth.txt: "leonidos:8JKPyY0sgCI+wCkoH9x"
2016-03-26 16:00:00: ERROR[main]: stack traceback:
2016-03-26 16:00:00: ERROR[main]:   [C]: in function 'error'
2016-03-26 16:00:00: ERROR[main]:   /home/quentinbd/mff-hg/bin/../builtin/game/auth.lua:48: in function 'read_auth_file'
2016-03-26 16:00:00: ERROR[main]:   /home/quentinbd/mff-hg/bin/../builtin/game/auth.lua:82: in main chunk
2016-03-26 16:00:00: ERROR[main]:   [C]: in function 'dofile'
2016-03-26 16:00:00: ERROR[main]:   /home/quentinbd/mff-hg/bin/../builtin/game/init.lua:19: in main chunk
2016-03-26 16:00:00: ERROR[main]:   [C]: in function 'dofile'
2016-03-26 16:00:00: ERROR[main]:   /home/quentinbd/mff-hg/bin/../builtin/init.lua:25: in main chunk
2016-03-26 16:00:00: ERROR[main]: ======= END OF ERROR FROM LUA ========
2016-03-26 16:00:01: ERROR[main]: ERROR: An unhandled exception occurred: ModError: Failed to load and run /home/quentinbd/mff-hg/bin/../builtin/init.lua
2016-03-26 16:00:01: ERROR[main]: Error from Lua:
2016-03-26 16:00:01: ERROR[main]: /home/quentinbd/mff-hg/bin/../builtin/game/auth.lua:48: Invalid line in auth.txt: "leonidos:8JKPyY0sgCI+wCkoH9x"
2016-03-26 16:00:01: ERROR[main]: stack traceback:
2016-03-26 16:00:01: ERROR[main]:   [C]: in function 'error'
2016-03-26 16:00:01: ERROR[main]:   /home/quentinbd/mff-hg/bin/../builtin/game/auth.lua:48: in function 'read_auth_file'
2016-03-26 16:00:01: ERROR[main]:   /home/quentinbd/mff-hg/bin/../builtin/game/auth.lua:82: in main chunk
2016-03-26 16:00:01: ERROR[main]:   [C]: in function 'dofile'
2016-03-26 16:00:01: ERROR[main]:   /home/quentinbd/mff-hg/bin/../builtin/game/init.lua:19: in main chunk
2016-03-26 16:00:01: ERROR[main]:   [C]: in function 'dofile'
2016-03-26 16:00:01: ERROR[main]:   /home/quentinbd/mff-hg/bin/../builtin/init.lua:25: in main chunk
----------------------
Server restarted at samedi 26 mars 2016, 16:00:26 (UTC+0100)
----------------------
Using game specified by --gameid on the command line
2016-03-26 16:00:26: ERROR[main]: ========== ERROR FROM LUA ===========
2016-03-26 16:00:26: ERROR[main]: Failed to load and run script from 
2016-03-26 16:00:26: ERROR[main]: /home/quentinbd/mff-hg/bin/../builtin/init.lua:
2016-03-26 16:00:26: ERROR[main]: /home/quentinbd/mff-hg/bin/../builtin/game/auth.lua:48: Invalid line in auth.txt: "leonidos:8JKPyY0sgCI+wCkoH9x"
2016-03-26 16:00:26: ERROR[main]: stack traceback:
2016-03-26 16:00:26: ERROR[main]:   [C]: in function 'error'
2016-03-26 16:00:26: ERROR[main]:   /home/quentinbd/mff-hg/bin/../builtin/game/auth.lua:48: in function 'read_auth_file'
2016-03-26 16:00:26: ERROR[main]:   /home/quentinbd/mff-hg/bin/../builtin/game/auth.lua:82: in main chunk
2016-03-26 16:00:26: ERROR[main]:   [C]: in function 'dofile'
2016-03-26 16:00:26: ERROR[main]:   /home/quentinbd/mff-hg/bin/../builtin/game/init.lua:19: in main chunk
2016-03-26 16:00:26: ERROR[main]:   [C]: in function 'dofile'
2016-03-26 16:00:26: ERROR[main]:   /home/quentinbd/mff-hg/bin/../builtin/init.lua:25: in main chunk
2016-03-26 16:00:26: ERROR[main]: ======= END OF ERROR FROM LUA ========
2016-03-26 16:00:26: ERROR[main]: ERROR: An unhandled exception occurred: ModError: Failed to load and run /home/quentinbd/mff-hg/bin/../builtin/init.lua
2016-03-26 16:00:26: ERROR[main]: Error from Lua:
2016-03-26 16:00:26: ERROR[main]: /home/quentinbd/mff-hg/bin/../builtin/game/auth.lua:48: Invalid line in auth.txt: "leonidos:8JKPyY0sgCI+wCkoH9x"
2016-03-26 16:00:26: ERROR[main]: stack traceback:
2016-03-26 16:00:26: ERROR[main]:   [C]: in function 'error'
2016-03-26 16:00:26: ERROR[main]:   /home/quentinbd/mff-hg/bin/../builtin/game/auth.lua:48: in function 'read_auth_file'
2016-03-26 16:00:26: ERROR[main]:   /home/quentinbd/mff-hg/bin/../builtin/game/auth.lua:82: in main chunk
2016-03-26 16:00:26: ERROR[main]:   [C]: in function 'dofile'
2016-03-26 16:00:26: ERROR[main]:   /home/quentinbd/mff-hg/bin/../builtin/game/init.lua:19: in main chunk
2016-03-26 16:00:26: ERROR[main]:   [C]: in function 'dofile'
2016-03-26 16:00:26: ERROR[main]:   /home/quentinbd/mff-hg/bin/../builtin/init.lua:25: in main chunk
----------------------
Server restarted at samedi 26 mars 2016, 16:00:51 (UTC+0100)
----------------------
```

I deleted the leonidas player account and the line in the auth.txt.
I hope it will resolve the issue.

**Comment by ElementW** [comment]
The skies are falling on our heads, even the core game has bugs...

**Comment by Lymkwi** [comment]
There is simply no way to fix that, it's a saving error from builtin. This has to be reported to minetest's repository if it keeps happening despite switching to a dev version of builtin and fixing that line.

**Comment by ObaniGemini** [comment]
You're lucky, it's the "issue-review" time at Minetest organization.

**Comment by BetterToAutomateTheWorld** [comment]
Can't be solved, hope 0.4.14-stable will solve this issue in the future :)

**Comment by ObaniGemini** [comment]
Added the upstream label. If we want this fixed some day, it should be reported upstream as @LeMagnesium said it.
